numba/cuda/cudadrv/driver.py

```diff
@@ -1735,10 +1751,18 @@ def __int__(self):
         return self.handle.value or 0
 
     def __repr__(self):
-        if self.handle.value:
-            return "<CUDA stream %d on %s>" % (self.handle.value, self.context)
+        default_streams = {
+            0: "<Default CUDA stream on %s>",
+            1: "<Legacy default CUDA stream on %s>",
+            2: "<Per-thread default CUDA stream on %s>",
+        }
+        ptr = self.handle.value or 0
+        if ptr in default_streams:
+            return default_streams[ptr] % self.context
+        elif self.external:
+            return "<External CUDA stream %d on %s>" % (ptr, self.context)
         else:
-            return "<Default CUDA stream on %s>" % self.context
+            return "<CUDA stream %d on %s>" % (ptr, self.context)
```

Comment:
hows this reached?

Reply:
By creating a non-default internal stream - e.g.:

```
In [3]: cuda.stream()                                                                                                                                        
Out[3]: <CUDA stream 94332750074080 on <CUDA context c_void_p(94332741571184) of device 0>>
```

Reply:
ah yes, thanks.